marlowe/src/Language/Marlowe/Core/V1/Semantics.hs

```diff
@@ -737,7 +739,7 @@ applyInput _ _ _ _                          = ApplyNoMatchError
 
 -- | Propagate 'ReduceWarning' to 'TransactionWarning'.
 convertReduceWarnings :: [ReduceWarning] -> [TransactionWarning]
-convertReduceWarnings = foldr (\warn acc -> case warn of
+convertReduceWarnings = foldr (\warn acc -> case warn of  -- Note that `foldr` is used here for efficiency, differing from Isabelle.
```

Comment:
Made a simple test to see how hard would it be to change this in isabelle. Early findings shows that some lemmas are broken. I will make a draft PR with a lot of `sorry` to see how big of an impact this will be